src/net_processing.cpp

```diff
@@ -752,6 +752,33 @@ void PeerLogicValidation::SyncTransaction(const CTransaction& tx, const CBlockIn
     }
 }
 
+void PeerLogicValidation::NewPoWValidBlock(const CBlockIndex *pindex, const std::shared_ptr<const CBlock>& pblock) {
+    CBlockHeaderAndShortTxIDs cmpctblock(*pblock, true);
+    CNetMsgMaker msgMaker(PROTOCOL_VERSION);
+
+    LOCK(cs_main);
+    bool fWitnessEnabled = IsWitnessEnabled(pindex->pprev, Params().GetConsensus());
+    uint256 hashBlock(pblock->GetHash());
```

Comment:
You can avoid calling it for each node by iterating mapNodeState for the set up front.

One PushMessage is no concern, but avoiding it for the entire ForEachNode would be nice.